src/GitHubExtension/Widgets/GitHubUserWidget.cs

```diff
@@ -57,15 +57,26 @@ public override void OnWidgetContextChanged(WidgetContextChangedArgs contextChan
 
     protected override void ResetWidgetInfoFromState()
     {
-        var dataObject = JsonNode.Parse(ConfigurationData);
+        try
+        {
+            var dataObject = JsonNode.Parse(ConfigurationData);
+
+            if (dataObject == null)
+            {
+                return;
+            }
 
-        if (dataObject == null)
+            ShowCategory = EnumHelper.StringToSearchCategory(dataObject["showCategory"]?.GetValue<string>() ?? string.Empty);
+            DeveloperLoginId = dataObject["account"]?.GetValue<string>() ?? string.Empty;
+        }
+        catch (Exception e)
         {
-            return;
+            // If we fail to parse configuration data, do nothing, report the failure, and don't
+            // crash the entire extension.
+            DeveloperLoginId = string.Empty;
+            ShowCategory = SearchCategory.Unknown;

```

Comment:
>SearchCategory.Unknown [](http://example.com/codeflow?start=27&length=22)

Just for my own understanding, what is the difference between `SearchCategory.Unknown` and `string.Empty`? (Intuitively, the choice here seems reasonable because we may not have gotten to the point of seeing what was in the configuration data due to an exception.)

Reply:
`string.Empty` falls back to `SearchCategory.Unknown`. So in this case, it is the same.

Reply:
Also, I chose those values because they were the default state of the variables when the class is created, with the assumption that it was a stable state to be in.